src/features/bivariate_color_manager/atoms/bivariateColorManagerResource.ts

```diff
@@ -227,3 +240,16 @@ export const bivariateColorManagerResourceAtom = createResourceAtom(
   null,
   true,
 );
+
+const fillLayersWithCorrelationLevel = (
+  bivariateColorManagerData: BivariateColorManagerData,
+  numeratorCorellationMap: NumeratorCorellationMap,
+): void => {
+  Object.values(bivariateColorManagerData).forEach((row) => {
+    [...Object.values(row.horizontal), ...Object.values(row.vertical)].forEach(
+      (layer) => {
+        layer.correlationLevel = numeratorCorellationMap[layer.name] || 0;
+      },
+    );
+  });
+};
```

Comment:
Just note:
Pretty unusual for me when function declared after they was used

Reply:
this approach is from https://marekhudyma.com/code-style/2021/03/02/step-down-rule.html

so when you open file, key and high level content is on top, some util functions that usually about details(low level) are below

Reply:
or sometimes I just unintentionally follow this rule
I'll put it on the top

Reply:
pretty interesting rule